src/dag-builder/file/flat.js

```diff
@@ -1,11 +1,11 @@
 'use strict'
 
-const batch = require('async-iterator-batch')
+const batch = require('it-flat-batch')
 
 module.exports = async function * (source, reduce) {
   const roots = []
 
-  for await (const chunk of batch(source, Infinity)) {
+  for await (const chunk of batch(source, Number.MAX_SAFE_INTEGER)) {
```

Comment:
Just use `it-all`?

Reply:
...I mean unless you actually want to receive a batch of `Number.MAX_SAFE_INTEGER` items at maximum 🤪

Reply:
Historically I think the assumption has been that `source` could return variable length arrays so we use a batcher that flattens them first, though the tests seem to pass with a regular batcher so I'm not sure that it's true any more.

In fact I think it only ever worked because of how `Array.concat` treats its arguments (e.g. not rejecting non-array arguments).

```
it-batch (internal buffer uses Array.push and yields when the buffer gets big enough)

input                         batcher                         output
child, child, child, child -> [child, child, child, child] -> [child, child, child, child] 
```

```
it-flat-batch (internal buffer uses Array.concat)

input                             batcher                         output
[child, child, child], [child] -> [child, child, child, child] -> [child, child, child, child] 
```

```
it-all (like it-batch but with an unbounded buffer size)

input                         output
child, child, child, child -> [child, child, child, child]
```

What fun.

TBH I don't know why we have this builder, it's not terribly useful and you can construct the same DAG with the balanced builder and a really high `maxChildrenPerNode` value.